internal/pkg/agent/application/upgrade/marker_access_test.go

```diff
@@ -95,6 +100,37 @@ func TestWriteMarkerFileWithTruncation(t *testing.T) {
 	close(errCh)
 }
 
+func TestUpdateMarkerFile(t *testing.T) {
+	marker := &UpdateMarker{
+		Version:           "1.2.3",
+		VersionedHome:     "/home",
+		Hash:              "sha...hash",
+		UpdatedOn:         time.Now(),
+		PrevVersion:       "0.1.2",
+		PrevHash:          "sha..hash",
+		PrevVersionedHome: "/home/elastic",
+		Acked:             false,
+		Action:            &fleetapi.ActionUpgrade{ActionID: "123", ActionType: "UPGRADAE"},
+		Details:           details.NewDetails("1.2.3", details.StateRequested, "action-id"),
+		DesiredOutcome:    OUTCOME_UPGRADE,
+	}
+	tmp := t.TempDir()
+	markerFile := filepath.Join(tmp, "marker")
+	require.NoError(t, saveMarkerToPath(marker, markerFile, true, true))
+
+	// update marker
+	err := UpdateMarkerFile(markerFile, func(m *UpdateMarker) {
+		m.Version = "1.2.3-up"
+	})
```

Comment:
Could we have 2 concurrent (i.e. different goroutines) partial updates on 2 different fields and verify that both updates apply cleanly?